### DISCUSSION TITLE
Is it nessecery to include a traditional code editor mode for editing?

### DISCUSSION THREAD

**Discussion by quank123wip** [Milkdown/milkdown#349] [Ideas]
As I'm interested in intergrating a non-compromised markdown-based rich-text editing environment for my frontend project, I have to make a solution for everybody, including people don't familiar with WYSIWYG editors, and prefer more traditional editing methods like they do in editors like Monaco Editor, Ace Editor, VSCode, etc. Is it a good idea to provide them as a plugin rather than let developers intergrate by themselves? It might be a hard work so I'm not asking for it.🤔

**Comment by Saul-Mirone** [comment]
To be honest. I think a code editor similar to codemirror or Monaco is out of scope for this editor. Because I want to solve the problems that haven't been solved by other editors. If we provide a code editor. I can't say how much better can we be compare to Monaco Editor or Codemirror.

And IMO, integrate a code editor is not a complicated work for users. All they need to do is just use listener plugin to get the markdown content of milkdown and send it to code editors. 

Let's leave it to users to decide which editor to choose to integrate. I believe we can have a good cooperation well with other editors.

**Reply by quank123wip** [reply]
Maybe it's true. It's not a complicated, anyone who know little in frameworks like react or vue can do it with ~100 lines of code. Even noobs like me can do it without pain. But for some less experienced people, it requires time to find out how to integrate them seamlessly and makes them look better. Maybe I will write.some poor examples or a plugin for them. But I'm not sure when I can do it. 😢If I really have it done, hope it can reduce your work on replying people who is interested in this project like me!

**Reply by Saul-Mirone** [reply]
I think a example in the `examples` folder can be really helpful. As for the plugin, I don't have plan to make it an official plugin, but I think it can be a good community plugin.
